### ISSUE TITLE
GOlr can no longer be loaded with annotations or bioentities

### ISSUE THREAD

**Issue by kltm** [geneontology/amigo#241]
Ugh, cannot get /anything/ loaded anymore:
```text
    While trying various reloads to track down the issue behind

    https://github.com/geneontology/amigo/issues/240

    I found two serious problems, both possibly related to owltools
    (although a schema error is possible for one). The first, and
    least severe, is that the load time has jumped from 4hrs to 14hrs
    for the "production" load, which is a huge hit. I believe that the
    memory required has also jumped up.
    
    The second thing, which is rather nasty, is that I am now no
    longer able to commit any GAF information, and the way the error
    is thrown, it effectively means that no annotation or bioentity
    data can get loaded.
    
    Since these have both come on rather suddenly, I wanted to warn 1)
    to avoid updating your tool chain until #240 is fixed (you should
    tag on to that item) and 2) check loads for annotation and
    bioentity information before pushing to production.
    
    I'll try to keep everybody up to date as we look at this.
```
Since I sent this email, I've reverted owltools to about a month ago (7919c6e) and still have the same problem, so I'm no longer going to look in that direction for the issue.

Next, schema (amigo) and java (???).

Tagging @stuartmiyasato, @cmungall, and @hdietze as this may affect the pipeline elsewhere.

Possible (temporal) relation to #240.

**Comment by stuartmiyasato** [comment]
I have run a new GOlr load but I have not copied the data anywhere.  With reference to Rama's ticket #245 , is it okay to push this load into production?  Should I push it to amigo2-test for validation first?  I'm holding off doing anything until I get feedback from someone...

**Comment by kltm** [comment]
There is a per-GAF fail like:
```
ERROR: [doc=UniProtKB:Q6ZMV9_:__:_GO:0001673_:__GO_REF:0000019_:_IEA_:__Ensembl:ENSRNOP00000015584_:_NCBITaxon:9606_:_20150711_:_Ensembl_:_] multiple values encountered for non multiValued field taxon: [NCBITaxon:9606, NCBITaxon:9606]
```
Occurs on commit, throwing an error that skips the rest of the GAF (i.e. all of it). Will check the proper value for the cardinality in the schemas and make sure there are no conflicts.

**Comment by cmungall** [comment]
Is this what @jelser was seeing?

**Comment by cmungall** [comment]
I mean @elserj

**Comment by stuartmiyasato** [comment]
@kltm :  just to be clear, this is happening during the GOlr load?  I'm not seeing anything like that in the load that I ran this morning.  My load finished without any errors, or at least I'm not seeing anything in the logs:

golr> grep -i error golr-load-20150908
Generating /data/src/svn/owltools/OWLTools-Oort/target/apidocs/owltools/ontologyrelease/logging/ErrorReportFileHandler.html...
Generating /data/src/svn/owltools/OWLTools-Oort/target/apidocs/owltools/ontologyrelease/logging/class-use/ErrorReportFileHandler.html...

I haven't copied those indexes anywhere.  Is it worth loading this data into amigo2-test to see if we're seeing the same issue?  Or am I looking at the wrong issue?

**Comment by kltm** [comment]
There are a few issues that have sort of piled up here. Personally, I am no longer able to load for an unknown reason (this issue); this may or may not be related to a couple of other issues.  Either way, this blocks me testing things like the "beta catenin" issue (#240), which is (possibly) a separate loading issue. Until I can make progress here, I'm uncomfortable proceeding with the 2.3 release.

**Comment by stuartmiyasato** [comment]
Is it worth comparing versions of the various software packages used in loading?  Or should I just hold off and wait until things are resolved on your end and then upgrade to whatever packages you are running?  Not meaning to be pushy here -- will certainly follow whatever recommendations you have.

**Comment by kltm** [comment]
I've already tried reverting to previous versions of owltools, but with no success. I'd wait to move forward until it's clear that this issue is local to me and won't "spread".

**Comment by kltm** [comment]
Well, not a direct schema mismatch, but still possible something is out of sync. Will try and get a minimal set that I can load fast (and fails) and will start tracking it through owltools.

**Comment by cmungall** [comment]
Do we have a mini test set? Either whole ontology + GAF subsets

Or if ontology size is an issue, we have some GAF subsets with matching 
ontology subset modules, in owltools test/resources folders

**Comment by kltm** [comment]
We do in the Makefiles, but even they are a bit irritating in tracking this down, which is why I'll be paring it down even more. I was hoping to find something a little more "realistic" (as I'm also keeping an eye out for #240), but I'll take a look at those if load times prevent fast turnaround.

**Comment by raymond91125** [comment]
in the off chance my experience helps here. i've seen similar problems for my wormbase build and it was caused by my solr server conf/schema.xml. i had to
# change schema.xml
<field name="taxon" type="string" required="false" multiValued="false" indexed="true" stored="true"></field>

to

<field name="taxon" type="string" required="false" multiValued="true" indexed="true" stored="true"></field>

I then restarted jetty and problem was gone.

**Comment by raymond91125** [comment]
yikes. magic github markdown

change schema.xml
```
<field name="taxon" type="string" required="false" multiValued="false" indexed="true" stored="true"></field>
to
<field name="taxon" type="string" required="false" multiValued="true" indexed="true" stored="true"></field>
```

**Comment by kltm** [comment]
Yes, that would fix the problem, but the issue is that it didn't do this in the past and /shouldn't/ be doing this now. When did you start needing to do this fix?

**Comment by raymond91125** [comment]
i think owltools
22dde7af7d6e65c1297c86d88dd80f7998e07818 # commit 05/27

**Comment by kltm** [comment]
Ah! Very interesting--thank you! That would mean that my test of month-old code would have missed it. I'll check in there!

https://github.com/owlcollab/owltools/commit/22dde7af7d6e65c1297c86d88dd80f7998e07818

Yes, this looks like a winner. Thank you again!

**Comment by cmungall** [comment]
@raymond91125 you rock!

Looks like ultimately it should be the java that changes and not the schema, the intent is for taxon to be single-valued

**Comment by hdietze** [comment]
Pushed a change. This appears to be the result of an incomplete refactor (to remove duplicate code) on my part.

**Comment by raymond91125** [comment]
Glad it helped.

Sent from my Cyanogen phone

On Sep 9, 2015 5:18 PM, hdietze <notifications@github.com> wrote:

Pushed a change. This appears to be the result of an incomplete refactor (to remove duplicate code) on my part.

**Comment by kltm** [comment]
Okay, Heiko's change probably is the fix, but a neat consequence is that we're likely to be forced to deal with #209 in the process as there is only one branch of owltools. This means that there will be schema changes as well and a little more testing necessary (as a new filter will be available, although we could pretend that it's not there for the time being).

Here's what we'll do, all current hotfixes, as well as #209 will be turned to the 2.3 milestone, fixed on master, and then bring the 2.3 up to date there.

I'll close this issue as soon as I have a confirmed proper load.

**Comment by kltm** [comment]
Confirmed a local load, closing for now.
